apiserver/facades/agent/upgradeseries/upgradeseries.go

```diff
@@ -282,6 +282,24 @@ func (a *API) authAndMachine(e params.Entity, canAccess common.AuthFunc) (common
 	return a.GetMachine(tag)
 }
 
+// PinnedLeadership returns all pinned applications and the entities that
+// require their pinned behaviour, for leadership in the current model.
+func (a *API) PinnedLeadership() (params.PinnedLeadershipResult, error) {
+	return a.leadership.PinnedLeadership()
+}
+
+// PinMachineApplications pins leadership for applications represented by units
+// running on the auth'd machine.
+func (a *API) PinMachineApplications() (params.PinApplicationsResults, error) {
+	return a.leadership.PinMachineApplications()
+}
+
+// UnpinMachineApplications unpins leadership for applications represented by
+// units running on the auth'd machine.
+func (a *API) UnpinMachineApplications() (params.PinApplicationsResults, error) {
+	return a.leadership.UnpinMachineApplications()
+}
+
 // disabledLeadershipPinningFacade implements the LeadershipPinningAPI, but
 // provides a no-operation for pinning operations
 type disabledLeadershipPinningFacade struct{}
```

Comment:
I am a fan of deleting this, but we might as well leave it in 2.7 and remove when we merge forward to where legacy leases have been removed altogether.

Reply:
Agreed.